packages/api/db/migration/20240115182150_create_animal_breed_tables.js

```diff
@@ -0,0 +1,130 @@
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+export const up = async function (knex) {
+  await knex.schema.createTable('default_animal_breed', (table) => {
+    table.increments('id').primary();
+    table.integer('default_type_id').notNullable().references('id').inTable('default_animal_type');
+    table.string('key').notNullable();
+  });
+
+  await knex.schema.createTable('custom_animal_breed', (table) => {
+    table.increments('id').primary();
+    table.uuid('farm_id').notNullable().references('farm_id').inTable('farm');
+    table.integer('default_type_id').references('id').inTable('default_animal_type');
+    table.integer('custom_type_id').references('id').inTable('custom_animal_type');
+    table.string('breed').notNullable();
+    table.boolean('deleted').notNullable().defaultTo(false);
+    table
+      .string('created_by_user_id')
+      .notNullable()
+      .references('user_id')
+      .inTable('users')
+      .defaultTo('1');
+    table
+      .string('updated_by_user_id')
+      .notNullable()
+      .references('user_id')
+      .inTable('users')
+      .defaultTo('1');
+    table.dateTime('created_at').notNullable().defaultTo(new Date('2000/1/1').toISOString());
+    table.dateTime('updated_at').notNullable().defaultTo(new Date('2000/1/1').toISOString());
+    table.check(
+      '?? is not null or ?? is not null',
+      ['default_type_id', 'custom_type_id'],
+      'not_null_type_id_check',
+    );
+    table.check(
+      '?? is null or ?? is null',
+      ['default_type_id', 'custom_type_id'],
+      'null_type_id_check',
+    );
+  });
+
+  // Update animal types table created_by_user_id and updated_by_user_id to match defaults used here
+  await knex.schema.alterTable('custom_animal_type', (table) => {
+    table.string('created_by_user_id').notNullable().defaultTo('1').alter();
+    table.string('updated_by_user_id').notNullable().defaultTo('1').alter();
+  });
+
+  // Add initial default breeds
+  const defaultBreedKeys = [
+    {
+      typeKey: 'CATTLE',
+      breedKeys: ['ANGUS', 'HEREFORD', 'CHAROLAIS'],
+    },
+    {
+      typeKey: 'PIGS',
+      breedKeys: ['YORKSHIRE_LARGE_WHITE', 'LANDRACE', 'DUROC'],
+    },
+    {
+      typeKey: 'CHICKEN_BROILERS',
+      breedKeys: ['CORNISH_CROSS', 'ROSS_308', 'COBB_500'],
+    },
+    {
+      typeKey: 'CHICKEN_LAYERS',
+      breedKeys: ['LEGHORN', 'RHODE_ISLAND_RED', 'PLYMOUTH_ROCK'],
+    },
+  ];
+
+  const rows = [];
+
+  for (const entry of defaultBreedKeys) {
+    const { typeKey, breedKeys } = entry;
+    const { id: typeId } = await knex('default_animal_type').where('key', typeKey).first();
```

Comment:
This line in the migration is failing for me on testing:db I wrote the reason for it below in the testEnvironment file.